exercises/all-your-base/test/test_all_your_base.c

```diff
@@ -0,0 +1,200 @@
+#include "vendor/unity.h"
+#include <stddef.h>
+#include <stdint.h>
+#include "../src/all_your_base.h"
+
+#include <stdio.h>
+
+void fill_array(int16_t data[], size_t data_length, int16_t array[],
+                size_t array_length)
```

Comment:
I realised I started with the right-justification because when initially designing the core logic the implementation didn't have the length of the input digits. With left-justification it wouldn't have known where to stop counting trailing zeros as digits ... but in the final implementation the function has the input digits length as an argument anyway so yes, the justification can be done away with.

As regards having the digit significance reversed.. I'm not sure without going in a bit more thoroughly but it would be moving away from the "what you see is what you get" idea a bit". 

I'll have a play around on both points this evening 👍